enhancements/sig-architecture/98-cluster-permission/README.md

```diff
@@ -0,0 +1,219 @@
+for RoleBinding and ClusterRoleBinding.
+- Allows the `ManagedServiceAccount` as Subject Kind 
+for RoleBinding and ClusterRoleBinding.
+
+### Future goals
+
+- It is currently assumed that the user of `ClusterPermission` is either a cluster admin or
+a user who can create `ClusterPermission` in the hub cluster's managed "cluster namespace".
+It will be useful if there is an approval process that a cluster admin can approve,
+which will allow for other users to create `ClusterPermission` resources.
+- Support creating a RoleBinding/ClusterRoleBinding without defining any Role/ClusterRole
+rules.
+
+## Design
+
+### Component & API
+
+We purpose to adding a new custom resource named
+`ClusterPermission` introduced into OCM by this proposal:
+
+A sample of the `ClusterPermission` will be:
+
+```yaml
+apiVersion: rbac.open-cluster-management.io/v1alpha1
+kind: ClusterPermission
+metadata:
+  name: my-managed-rbac
+  namespace: cluster1
+spec:
+  clusterRole:
+    rules:
+    - apiGroups: ["apps"]
+      resources: ["deployments"]
+      verbs: ["update"]
+  roles:
+  - namespace: default
+    rules:
+    - apiGroups: ["apps"]
+      resources: ["deployments"]
+      verbs: ["update"]
+  - namespaceSelector:
+      matchExpressions:
+      - key: foo.com/managed-state
+        operator: In
+        values:
+        - managed
+    rules:
+    - apiGroups: [""]
+      resources: ["configmaps"]
+      verbs: ["update"]
+  subject:
+    apiGroup: authentication.open-cluster-management.io
+    kind: ManagedServiceAccount
+    name: managed-sa-sample
+status:
+  conditions:
+    - type: RBACManifestWorkCreated
+      status: True
+    - type: SubjectPresent
+      status: True    
+```
+
+The `ClusterPermission` resource is expected to be created under the
+"cluster namespace" which is a namespace with the same name as
+the managed cluster, the Role/RoleBinding/ClusterRole/ClusterRoleBinding
+delivered to the managed cluster will have the same name as the
+`ClusterPermission` resource.
+
+We purpose adding a new hub cluster controller that watches
+and reconcile `ClusterPermission`. When `ClusterPermission` is reconciled, create/update
+a `ManifestWork` with the payload of Roles, ClusterRole,
+RoleBindings, and ClusterRoleBinding. All the RBAC resources will have
+the same name as the `ClusterPermission`. The `ManifestWork` owner is set to 
+`ClusterPermission` so that when the  `ClusterPermission` is deleted, the `ManifestWork` 
+will be garbage collected, which will trigger the removal of all the RBAC 
+related resources in the managed cluster. It's expected that the executor 
+of the `ManifestWork` is the work agent which has privilege to create
+RBAC resources on the managed cluster. Therefore, the spec of the `ManifestWork`
+will hard the klusterlet-work-sa service account hardcoded:
+
+```
+  executor:
+    subject:
+      type: ServiceAccount
+      serviceAccount:
+        namespace: open-cluster-management-agent
```

Comment:
nit, for hosted mode klusterlet, the sa is not in the `open-cluster-management-agent` ns.

Reply:
Thank you @zhujian7 for the great comment. Do you have any recommendation what we should do in the hosted mode case?

Reply:
I'd suggest keeping the executor as nil, but granting the "clusterPermission-controller-serviceaccount" permission to [use the `system:serviceaccount::klusterlet-work-sa` as the executor](https://github.com/open-cluster-management-io/api/blob/54a83a1650d6aa62f4bfe14b719a88a9f191b05a/feature/feature.go#L49-L50)

because if the executor is nil, the work-webhook will [assume it is using the klusterlet-work-sa as the executor](https://github.com/open-cluster-management-io/work/blob/main/pkg/webhook/v1/manifestwork_validating.go#L83-L94).

Reply:
Updated enhancement based on Zhu Jian's recommendations.